### DISCUSSION TITLE
Add Rows and Columns properties to UniformGrid

### DISCUSSION THREAD

**Discussion by DoubleDBE** [CommunityToolkit/Maui#156] [New Feature Discussions]
Coming from [#122 ](https://github.com/CommunityToolkit/Maui/issues/122#issuecomment-945168116)

In WPF, we have the ability to specify how much [rows](https://docs.microsoft.com/en-us/dotnet/api/system.windows.controls.primitives.uniformgrid.rows?view=windowsdesktop-5.0#System_Windows_Controls_Primitives_UniformGrid_Rows) and [columns](https://docs.microsoft.com/en-us/dotnet/api/system.windows.controls.primitives.uniformgrid.columns?view=windowsdesktop-5.0#System_Windows_Controls_Primitives_UniformGrid_Columns) the `UniformGrid` must generate.

eg `<UniformGrid Rows="1" Columns="2"/>`

This gives us more control on how the `UniformGrid` should behave.

As explained in the WPF documentation:
`A value of zero (0) for the Rows property specifies that the row count is computed based on the number of columns and the number of visible child elements that are in the grid.`

`A value of zero (0) for the Columns property specifies that the column count is computed based on the number of rows and the number of visible child elements that are in the Grid.`

This means that by adding these properties, the current behaviour won't be affected and developers coming from XCT won't have any issues porting over to MCT since the `UniformGrid` should behave as-is when these properties are not specified.

For completeness sake, we could also add the [FirstColumn](https://docs.microsoft.com/en-us/dotnet/api/system.windows.controls.primitives.uniformgrid.firstcolumn?view=windowsdesktop-5.0#System_Windows_Controls_Primitives_UniformGrid_FirstColumn) property although i've never used that property ever before.

**Comment by VladislavAntonyuk** [comment]
I suggest renaming it to MaxRows and MaxColumns. by default both values contain int.MaxValue. if the value is set for any of these properties we will calculate based on the value

**Comment by VladislavAntonyuk** [comment]
You can track implementation in this PR: https://github.com/CommunityToolkit/Maui/pull/139

**Comment by DoubleDBE** [comment]
Awesome!

**Comment by CliffAgius** [comment]
Closed as merged.
